WPFHospitalEditor/RoomScheduledExaminations.xaml.cs

```diff
@@ -0,0 +1,162 @@
+            if (FindAndDeleteByExaminationId(roomRelocations, examinationId)) return;
+            FindAndDeleteByExaminationId(roomRenovations, examinationId);
+        }
+
+        private bool FindAndDeleteByExaminationId(ObservableCollection<RoomScheduledAppointmentDto> rsaDtos, int examinationId)
+        {
+            foreach (RoomScheduledAppointmentDto rsaDto in rsaDtos)
+            {
+                if (rsaDto.ExaminationId == examinationId)
+                {
+                    rsaDtos.Remove(rsaDto);
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private void FillObservableCollection()
+        {
+            CreateEmptyObservableCollections();
+            List<Examination> allRoomAppointments = examinationServerController.getByRoomId(roomId).ToList();
+            foreach (Examination examination in allRoomAppointments)
+            {
+                if(!examination.IsCanceled)
+                {
+                    RoomScheduledAppointmentDto rsaDto = CreateRoomScheduledAppointmentDto(examination);
+                    if (examination.PatientId == AllConstants.PatientIdForRelocation)
+                        roomRelocations.Add(rsaDto);
+                    else if (examination.PatientId == AllConstants.PatientIdForRenovation)
+                        roomRenovations.Add(rsaDto);
+                    else
+                        roomExaminations.Add(rsaDto);
+                }
+            }
+        }
+
+        private RoomScheduledAppointmentDto CreateRoomScheduledAppointmentDto(Examination examination)
+        {
+            RoomScheduledAppointmentDto rsaDto;
+            rsaDto = new RoomScheduledAppointmentDto
+            {
+                ExaminationId = examination.Id,
+                StartTime = examination.TimeInterval.Start,
+                EndTime = examination.TimeInterval.End,
+                DoctorId = examination.DoctorId,
+                PatientId = examination.PatientId
+            };
+            return rsaDto;
+        }
+
+        private void CreateEmptyObservableCollections()
+        {
+            roomExaminations = new ObservableCollection<RoomScheduledAppointmentDto>();
+            roomRelocations = new ObservableCollection<RoomScheduledAppointmentDto>();
+            roomRenovations = new ObservableCollection<RoomScheduledAppointmentDto>();
+        }
+
+        private void AppointmentTypeComboBoxSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if(scrollViewer1 != null)
+            {
+                if (appointmentTypeComboBox.SelectedIndex == 0)
+                {
+                    scrollViewer1.Visibility = Visibility.Visible;
+                    scrollViewer2.Visibility = Visibility.Collapsed;
+                    scrollViewer3.Visibility = Visibility.Collapsed;
+                }
+                else if (appointmentTypeComboBox.SelectedIndex == 1)
```

Comment:
I will collapse all scrolls on start and just put visibility on for scroll that is needed. It will reduce number of lines.

Reply:
But, if we change selected item in combobox, we must collapse other scrolls.

Reply:
Fixed.